cmd/neo4j-mcp/main.go

```diff
@@ -83,17 +87,33 @@ func main() {
 	// Create and configure the MCP server
 	mcpServer := server.NewNeo4jMCPServer(Version, cfg, dbService, anService)
 
-	// Gracefully handle shutdown
-	defer func() {
-		if err := mcpServer.Stop(); err != nil {
-			slog.Error("Error stopping server", "error", err)
+	if cfg.TransportMode == config.TransportModeHTTP {
```

Comment:
I would handle this directly  in the Server.Stop method, I think the same benefits can be also be applied with the STDIO, and both can be harmonised.
But I'm interested also in your opinion, what do you think?
The place where I'm coming from is the desire to keep the `main func` simple and easier to read while giving the server full ownership of signal handling.

Reply:
I think it’s a good point and I generally agree to keeping main simple, although it’s a common pattern for servers to have a sigterm listener in main that blocks until the exit signal is received.

I also remember there was not much to stop in the stdio version, I mean it was a blank function if I remember correctly but I will check this again in the morning:)